iroh-net/src/lib.rs

```diff
@@ -4,6 +4,39 @@
 //! interface to [QUIC] connections and streams to the user, while implementing direct
 //! connectivity using [hole punching] complemented by relay servers under the hood.
 //!
+//! Connecting to a remote node looks roughly like this:
+//!
+//! ```no_run
+//! # use iroh_net::{Endpoint, NodeAddr};
+//! # async fn wrapper() -> testresult::TestResult {
+//! let addr: NodeAddr = todo!();
+//! let ep = Endpoint::builder().bind().await?;
+//! let conn = ep.connect(addr, b"my-alpn").await?;
+//! let mut send_stream = conn.open_uni().await?;
+//! send_stream.write_all(b"msg").await?;
+//! # Ok(())
+//! # }
+//! ```
+//!
+//! Accepting connections is a little more work:
+//!
+//! ```no_run
+//! # use iroh_net::{Endpoint, NodeAddr};
+//! # async fn wrapper() -> testresult::TestResult {
+//! let ep = Endpoint::builder()
+//!     .alpns(vec![b"by-alpn".to_vec()])
+//!     .bind()
+//!     .await?;
+//! let conn = ep.accept().await.ok_or("err")?.await?;
+//! let mut recv_stream = conn.accept_uni().await?;
+//! let mut buf = [0u8; 3];
+//! recv_stream.read_exact(&mut buf).await?;
+//! # Ok(())
+//! # }
+//! ```
+//!
+//! Of course you can also use [bi-directional streams] or any other features from QUIC.
+//!
```

Comment:
I see you've kept the more fully-featured example below. So perhaps we link to it?

```suggestion
//! Of course you can also use [bi-directional streams] or any other features from QUIC.
//!
//! For a more elaborate examples, see [below](#examples) or the examples directory in
//! the source repository.
//!
```

wdyt?

Reply:
yeah, i nearly added that as well